crates/core/src/rpc/ws.rs

```diff
@@ -193,4 +228,120 @@ impl Rpc for SurfpoolWsRpc {
             })
         }
     }
+
+    fn account_subscribe(
+        &self,
+        meta: Self::Metadata,
+        subscriber: Subscriber<RpcResponse<UiAccount>>,
+        pubkey_str: String,
+        config: Option<RpcAccountSubscribeConfig>,
+    ) {
+        let pubkey = match Pubkey::from_str(&pubkey_str) {
+            Ok(pk) => pk,
+            Err(_) => {
+                let error = Error {
+                    code: ErrorCode::InvalidParams,
+                    message: "Invalid pubkey format.".into(),
+                    data: None,
+                };
+                subscriber.reject(error.clone()).unwrap();
+                return;
+            }
+        };
+
+        let config = config.unwrap_or(RpcAccountSubscribeConfig {
+            commitment: None,
+            encoding: None,
+        });
+
+        let id = self.uid.fetch_add(1, atomic::Ordering::SeqCst);
+        let sub_id = SubscriptionId::Number(id as u64);
+        let sink = subscriber
+            .assign_id(sub_id.clone())
+            .expect("Failed to assign subscription ID");
+
+        let account_active = Arc::clone(&self.account_active);
+        let meta = meta.clone();
+
+        self.tokio_handle.spawn(async move {
+            account_active.write().unwrap().insert(sub_id.clone(), sink);
+
+            let SurfnetRpcContext {
+                svm_locker,
+                remote_ctx,
+            } = match meta.get_rpc_context(CommitmentConfig::confirmed()) {
+                Ok(res) => res,
+                Err(_) => panic!(),
+            };
+
+            // get the account from the SVM to see if it exists
+            let result = svm_locker.get_account(&remote_ctx, &pubkey, None).await;
+
+            if let Ok(SvmAccessContext {
+                inner: account_result,
+                ..
+            }) = result
+            {
+                // if we found the account, send the initial notification
+                if let GetAccountResult::FoundAccount(_, account, _) = account_result {
+                    let ui_account = encode_ui_account(
+                        &pubkey,
+                        &account,
+                        config.encoding.unwrap_or(UiAccountEncoding::Base64),
+                        None,
+                        None,
+                    );
+                    if let Some(sink) = account_active.read().unwrap().get(&sub_id) {
+                        let _ = sink.notify(Ok(RpcResponse {
+                            context: RpcResponseContext::new(
+                                svm_locker.with_svm_reader(|svm| svm.get_latest_absolute_slot()),
+                            ),
+                            value: ui_account,
+                        }));
+                    }
+                }
+            }
+
+            // subscribe to account updates
+            let rx = svm_locker.subscribe_for_account_updates(&pubkey, config.encoding);
+
+            loop {
```

Comment:
We need to be able to break out of this loop. Currently, if you never receive an account update, you'll never break out of the loop, right?

I think we need something like:
```rust
if let Some(sink) = account_active.read().unwrap().get(&sub_id).is_none() {
     break;
}
```
after each `rx.try_recv()`